types/jss/index.d.ts

```diff
@@ -1,50 +1,124 @@
-// Type definitions for jss v0.6
-// Project: https://github.com/Box9/jss
-// Definitions by: Valentin Robert <https://github.com/Ptival>
+// Type definitions for jss 9.3
+// Project: https://github.com/cssinjs/jss#readme
+// Definitions by: Brenton Simpson <https://github.com/appsforartists>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.2
 
-interface Properties {
-    [name: string]: string;
+export interface Rule {
+	className: string;
+	selector: string;
+	applyTo(element: HTMLElement): void;
+	prop(key: string, value?: any): Rule | string;
+	toJSON(): string;
 }
-
-interface Selectors {
-    [selector: string]: Properties;
+export interface StyleSheet<T> {
+	// Gives auto-completion on the rules declared in `createStyleSheet` without
+	// causing errors for rules added dynamically after creation.
+	classes: {
+		[K in keyof T]: string;
+	} & { [key: string]: string };
+	options: any;
+	linked: boolean;
+	attached: boolean;
+	/**
+	 * Attach renderable to the render tree.
+	 */
+	attach(): this;
+	/**
+	 * Remove renderable from render tree.
+	 */
+	detach(): this;
+	/**
+	 * Add a rule to the current stylesheet.
+	 * Will insert a rule also after the stylesheet has been rendered first time.
+	 */
+	addRule(style: Style, options?: Partial<RuleOptions>): Rule;
+	addRule(name: string, style: Style, options?: Partial<RuleOptions>): Rule;
+	/**
+	 * Create and add rules.
+	 * Will render also after Style Sheet was rendered the first time.
+	 */
+	addRules(styles: { [key: string]: Style }, options?: Partial<RuleOptions>): Rule[];
+	/**
+	 * Get a rule by name.
+	 */
+	getRule(name: keyof T | string): Rule;
+	/**
+	 * Delete a rule by name.
+	 * Returns `true`: if rule has been deleted from the DOM.
+	 */
+	deleteRule(name: string): boolean;
+	/**
+	 * Get index of a rule.
+	 */
+	indexOf(rule: Rule): number;
+	/**
+	 * Update the function values with a new data.
+	 */
+	update(data?: {}): this;
+	update(name?: string, data?: {}): this;
```

Comment:
These parameters don't need to be optional since the ability to pass 0 or 1 arguments is handled by the first overload